src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.xml

```diff
@@ -66,6 +67,7 @@
                 <argument key="matcher_base_class">%router.options.matcher_base_class%</argument>
                 <argument key="matcher_dumper_class">%router.options.matcher_dumper_class%</argument>
                 <argument key="matcher_cache_class">%router.options.matcher.cache_class%</argument>
+                <argument key="route_collection_cache_file">%router.options.route_collection.cache_file%</argument>
```

Comment:
@nicolas-grekas,  could you explain a little bit more about your suggestion? I don't understand how to reload the route collection when one of the related resources is changed. 
I just copied the logic from a `getGenerator` method https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Routing/Router.php#L318